Jenkinsfile

```diff
@@ -40,11 +42,25 @@ pipeline {
         HTTP_DIGEST_ACL=credentials("portaljs.${env.CF_SPACE}.http.digest.acl")
       }
       steps {
+        sh 'if [[ "${LOGSTASH_CONNECTION}" != "" ]]; then echo "services:\\n  - {CF_LOG_DRAINER_SERVICE}" >> manifest.yml; fi'
         sh 'sed -i "s|env:|env:\\n  CTF_CPA_ACCESS_TOKEN: ${CTF_CPA_ACCESS_TOKEN}|" manifest.yml'
         sh 'sed -i "s|env:|env:\\n  HTTP_DIGEST_ACL: ${HTTP_DIGEST_ACL}|" manifest.yml'
         sh 'cf blue-green-deploy ${CF_APP_NAME} -f manifest.yml --delete-old-apps'
       }
     }
+    stage('Register with ELK') {
+      when {
+        expression { return env.LOGSTASH_CONNECTION }
+      }
+      environment {
+        CF_APP_NAME="portaljs${env.CF_SPACE == 'production' ? '' : '-' + env.CF_SPACE}"
```

Comment:
Move this up to the top level environment instead, along with the other declaration of it.

Reply:
done, although this assignment depends on the space being set which also happens in that block, so I hope this doesn't cause any issues.